FORMAT.md

```diff
@@ -1,40 +1,99 @@
+# NOCK Open Standard
+
 [Apache Arrow](https://arrow.apache.org/docs/index.html) 
 and its [Parquet](https://arrow.apache.org/docs/cpp/parquet.html) format
 provide the most efficient means for graph serialization and persistence.
 
+This proposed `NOCK` open standard serializes graphs efficiently at
+scale in a way which aligns the data representations required for
+popular graph technologies and related data sources:
+
+  * semantic graphs (e.g., W3C formats RDF, TTL, JSON-LD, etc.)
+  * labeled property graphs (e.g., openCypher)
+  * probabilistic graphs (e.g., PSL)
+  * spreadsheet import/export (e.g., CSV)
+  * dataframes (e.g., Pandas, Dask, Spark, etc.)
+  * edge lists (e.g., NetworkX, cuGraph, etc.)
+
+
+## Terminology
+
+Graph data has two possible states:
+
+  * _marshalled_: serialized and persisted in storage, i.e., "at rest"
+  * _unmarshalled_: dynamic data structures in memory, i.e., "live"
+
+
+A node may be referenced either as a _source node_, which has directed edges, or as a _destination node_ which is the target of an edge.
+
+When a node from another partition is referenced as a _destination node_, then at least its "shadow" information (i.e., its unique symbol) gets included within the referencing partition. This is called a _shadow node_.
+
+When a shadow node gets unmarshalled, that triggers an `asyncio` _future_ (called an _object reference_ in Ray) to perform a distributed lookup of the node by name across the cluster. Then its partition info replaces the `"edge_id"` value.
+
+
+## Conventions: Missing Values, etc.
+
+Data frameworks such as Excel and `pandas` have conflicting rules and default settings for how to handle missing values when marshalling and unmarshalling data. Language differences (Python, C++, SQL) as well as their popular libraries for handling CSV, JSON, dataframes, and so on, impose their own rules in addition. Consequently we encounter a range of possible ways to represent missing values:
+
+  * `""` (empty string)
+  * `NA`
+  * `NaN`
+  * `None`
+  * `null`
+
+Therefore to help minimize data quality surprises, `NOCK` uses the following missing values for the sake of improved consistency:
+
+  * integer columns: `-1`
```

Comment:
Should this also apply to `float`? (e.g. the `truth` column)

Reply:
Hmm, I'm not familiar with the use case for `truth` column, so my prior is that it's possible to have missing values for this (and this might be distinct from 0 truth value).

Reply:
I've added validation checkers for this